### PR TITLE
[Store onboarding] Add `blog_public` to `Site`

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #9477

## Description

Dashboard with "Launch your store" task

The "Launch your store" onboarding task is displayed as incomplete for stores that are already live. Many users have reported that this is confusing as their store has been launched a long time back. 

We inspected the problem, and it is an issue with the backend. Internal discussion -  p1681200289275869-slack-CNA89KY6M

While the issue is being fixed in the backend, we decided to release a workaround fix to hide the "Launch your store" task for already live sites. We are going to use the `blog_public` value from the `me/sites` response. Internal discussion - p1681726337528639-slack-C045CUK1Y3U

This PR,
- Adds `blog_public` property to `Site`. (Updates both Networking and Storage layers)
- Adds new core data model version and tests migration.
- Starts getting `blog_public` from `me/sites/` request from `AccountRemote`

## Testing instructions

1. Launch the app, and log out if needed.
2. Login back, and confirm that the store list is loaded correctly.
3. Select a store, and confirm it loads correctly.

## Screenshots

NA

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/Networking/Model/Site.swift [hunk 1]
```diff
@@ -64,6 +64,10 @@ public struct Site: Decodable, Equatable, GeneratedFakeable, GeneratedCopiable {
     ///
     public let gmtOffset: Double
 
+    /// Whether the site is launched and public.
+    ///
+    public let isBlogPublic: Bool
```
Comment (jaclync) [inline Networking/Networking/Model/Site.swift]: nit: "blog" is more for the WordPress blogs, we can drop the blog or refer to "site"
```suggestion
    public let isPublic: Bool
```

File: Storage/Storage/Model/Site+CoreDataProperties.swift [hunk 2]
```diff
@@ -22,6 +22,7 @@ extension Site {
     @NSManaged public var isJetpackConnected: Bool
     @NSManaged public var isJetpackThePluginInstalled: Bool
     @NSManaged public var jetpackConnectionActivePlugins: [String]?
+    @NSManaged public var isBlogPublic: Bool
```
Comment (jaclync) [inline Storage/Storage/Model/Site+CoreDataProperties.swift]: nit: similar to the renaming comment above, "blog" sounds confusing in the WooCommerce context
```suggestion
    @NSManaged public var isPublic: Bool
```

**Review by jaclync — APPROVED** [review]
LGTM, just a naming suggestion.

I'd also recommend notifying the WCiOS team about the Core Data schema change in the release branch, in case anyone is also planning to change the schema in `trunk`.

**Review by jostnes — APPROVED** [review]
thanks for updating the mock file to match the latest change! i'm only reviewing https://github.com/woocommerce/woocommerce-ios/pull/9480/commits/fd147828e700e2e77f3907032571e5e51a88b297 commit for the UI test change, and the changes look good! i see tests are passing in CI ✅

File: Storage/Storage/Model/Site+CoreDataProperties.swift (hunk 2 above)
Comment (selanthiraiyan) [reply Storage/Storage/Model/Site+CoreDataProperties.swift]: I agree. Renamed in https://github.com/woocommerce/woocommerce-ios/pull/9480/commits/a8b9dd21252ce9faf6cb80f2e528f46e5040e3d8

File: Networking/Networking/Model/Site.swift (hunk 1 above)
Comment (selanthiraiyan) [reply Networking/Networking/Model/Site.swift]: Renamed in https://github.com/woocommerce/woocommerce-ios/pull/9480/commits/a8b9dd21252ce9faf6cb80f2e528f46e5040e3d8

**Comment by selanthiraiyan** [comment]
@oguzkocer :wave:  Heads-up about this beta fix PR

* Please release a beta build once this gets merged.
* Also, please merge these code changes back to develop, as this contains changes to the core data schema. 

Thank you! :bow:

**Comment by oguzkocer** [comment]
> @oguzkocer 👋 Heads-up about this beta fix PR
> Please release a beta build once this gets merged.
> Also, please merge these code changes back to develop, as this contains changes to the core data schema.

cc @spencertransier

**Pull request merged**